src/app/clusters/operational-credentials-server/operational-credentials-cluster.cpp

```diff
@@ -937,452 +665,683 @@ bool emberAfOperationalCredentialsClusterUpdateNOCCallback(app::CommandHandler *
+                                                                             Commands::UpdateNOC::kMetadataEntry,
+                                                                             Commands::UpdateFabricLabel::kMetadataEntry,
+                                                                             Commands::RemoveFabric::kMetadataEntry,
+                                                                             Commands::AddTrustedRootCertificate::kMetadataEntry,
+                                                                             Commands::SetVIDVerificationStatement::kMetadataEntry,
+                                                                             Commands::SignVIDVerificationRequest::kMetadataEntry };
+
+    return builder.ReferenceExisting(kAcceptedCommands);
+}
 
-    if (!IsValidFabricIndex(commandData.fabricIndex) ||
-        (commandData.clientChallenge.size() != kVendorIdVerificationClientChallengeSize))
+CHIP_ERROR OperationalCredentialsCluster::GeneratedCommands(const ConcreteClusterPath & path,
+                                                            ReadOnlyBufferBuilder<CommandId> & builder)
+{
+    static constexpr CommandId kGeneratedCommands[] = { Commands::AttestationResponse::Id, Commands::CertificateChainResponse::Id,
+                                                        Commands::CSRResponse::Id, Commands::NOCResponse::Id,
+                                                        Commands::SignVIDVerificationResponse::Id };
+
+    return builder.ReferenceExisting(kGeneratedCommands);
+}
+
+std::optional<DataModel::ActionReturnStatus> OperationalCredentialsCluster::InvokeCommand(const DataModel::InvokeRequest & request,
+                                                                                          TLV::TLVReader & input_arguments,
+                                                                                          CommandHandler * handler)
+{
+    switch (request.path.mCommandId)
     {
-        commandObj->AddStatus(commandPath, Status::ConstraintError);
-        return true;
+    case OperationalCredentials::Commands::AttestationRequest::Id:
+        return HandleAttestationRequest(handler, request.path, input_arguments, GetDACProvider());
+    case OperationalCredentials::Commands::CertificateChainRequest::Id:
+        return HandleCertificateChainRequest(handler, request.path, input_arguments, GetDACProvider());
+    case OperationalCredentials::Commands::CSRRequest::Id:
+        return HandleCSRRequest(handler, request.path, input_arguments, GetFabricTable(), GetFailSafeContext(), GetDACProvider());
+    case OperationalCredentials::Commands::AddNOC::Id: {
+        std::optional<DataModel::ActionReturnStatus> returnStatus =
+            HandleAddNOC(handler, request.path, input_arguments, GetFabricTable(), GetFailSafeContext(), GetDNSSDServer(),
+                         GetCommissioningWindowManager());
+        if (returnStatus == Status::Success)
```

Comment:
ActionReturnStatus in theory has many successes: CHIP_ERROR, cluster success status or Status::Success.

What I believe is happening here is that you create an ActionReturnStatus out of Status::Success and then use operator== .... could we use `returnStatus.IsSuccess()` instead?